packages/editor/src/hooks/align.js

```diff
@@ -23,6 +23,10 @@ import { BlockControls, BlockAlignmentToolbar } from '../components';
  * @return {Object}          Filtered block settings
  */
 export function addAttribute( settings ) {
+	// allow blocks to specify their own attribute definition with default values if needed.
+	if ( get( settings.attributes, [ 'align', 'type' ] ) === 'string' ) {
```

Comment:
Why do we specifically check the type? Would it be okay for us to override the attribute definition if it was a type other than `string` ? Would `has( settings.attributes, [ 'align' ] )` not have been sufficient?

Reply:
We are checking the type because when using the align hook the hook is the one setting the align value and the hook sets a string, so specifying an attribute with a different type would not work correctly.
When setting supports align the expectation is that the hook will work properly so in this case for the hook to work properly we need to change the align attribute, I think it is acceptable.